test/app-luatest/key_estimator_test.lua

```diff
@@ -0,0 +1,110 @@
+local t = require('luatest')
+local g = t.group()
+local key_def = require('key_def')
+local key_estimator = require('key_estimator')
+
+local function hll_error(prec)
+    return 1.04 / (2 ^ (prec / 2.0))
+end
+
+g.test_constants = function()
+    t.assert_not_equals(key_estimator.SPARSE, key_estimator.DENSE)
+    t.assert_lt(key_estimator.MIN_PRECISION, key_estimator.MAX_PRECISION)
+end
+
+g.test_common = function()
+    --- expect key format
+    t.assert_error(key_estimator.new)
+
+    local fmt = key_def.new({{type = "string", fieldno = 1}})
+    local est = key_estimator.new(fmt)
+
+    --- invalid precision
+    t.assert_error(key_estimator.new, fmt, key_estimator.MAX_PRECISION + 1)
+
+    t.assert_equals(est:estimate(), 0)
+    est:add({'bla'})
+    t.assert_equals(est:estimate(), 1)
+    est:add({'bla'})
+    t.assert_equals(est:estimate(), 1)
+    est:add({'bla', 'bla', 'bla..'})
+    t.assert_equals(est:estimate(), 1)
+
+    --- wrong key format, string is expected
+    t.assert_error(key_estimator.add, est, {0})
+end
+
+g.test_error = function()
+    local fmt = key_def.new({{type = "string", fieldno = 1}})
+
+    local min_prec = key_estimator.MIN_PRECISION
+    local max_prec = key_estimator.MAX_PRECISION
+    for prec = min_prec, max_prec, 1 do
+        local est = key_estimator.new(fmt, prec)
+        local err = hll_error(prec)
+
+        local card = 10 * 2 ^ prec
```

Comment:
Please supply this formula with a comment.

Reply:
Done.